packages/main/src/RPA/Smartsheet.py

```diff
@@ -0,0 +1,1415 @@
+            elif key in ["rowNumber", "row_number"]:
+                row.id = self._get_row_from_number(value)
+            else:
+                cell.column_id = self._get_column_id(key)
+                cell.value = value
+                row.cells.append(cell)
+        return row
+
+    def _get_row_from_number(self, row_number: int) -> int:
+        """Returns the row ID from a row number. If not found, it
+        assumes the provided row_number may have been an ID and
+        returns it.
+        """
+        self._require_current_sheet()
+        for row in self.current_sheet.rows:
+            if row.row_number == row_number:
+                return row.id
+        return row_number
+
+    def _create_row_from_list(self, row_list: List[Dict]) -> Row:
+        """Creates a row object from a list."""
+        row = Row()
+        for cell_dict in row_list:
+            if cell_dict.get("title") == "rowId":
+                row.id = cell_dict.get("value")
+            elif cell_dict.get("title") == "rowNumber":
+                row.id = self._get_row_from_number(cell_dict.get("value"))
+            else:
+                cell = self._convert_dict_to_cell(cell_dict)
+                row.cells.append(cell)
+        return row
+
+    def _convert_data_to_row(self, data: Union[Dict, List, Row]) -> Row:
+        """Converts a dictionary or list to a row object."""
+        if isinstance(data, dict):
+            return self._create_row_from_dict(data)
+        elif isinstance(data, list):
+            return self._create_row_from_list(data)
+        elif isinstance(data, Row):
+            return data
+        else:
+            raise TypeError(
+                f"Invalid row type. Received {type(data)}, expected dict, list or Row."
+            )
+
+    @keyword
+    def convert_row_to_dict(self, row: Row) -> OrderedDict:
+        """Converts a row object to a dictionary."""
+        row_dict = OrderedDict()
+        for cell in row.cells:
+            row_dict[cell.column_id] = cell.value
+        return row_dict
+
+    @keyword
+    def get_row(
+        self,
+        row: Union[int, Row],
+        include: CommaListType = None,
+        exclude: CommaListType = None,
+        native: bool = False,
+    ) -> Row:
+        """Returns a single row from the current sheet.
+
+        You can provide the row as a native ``Row`` object or as an
+        integer representing the row ID.
+        """
+        self._require_current_sheet()
+        include = self._parse_include(self.ROW_INCLUDES, include)
+        exclude = self._parse_exclude(self.ROW_EXCLUDES, exclude)
+        row_id = self._parse_row_id(row)
+        row = self.smart.Sheets.get_row(self.current_sheet.id, row_id)
+        if native:
+            return row
+        else:
+            return self.convert_row_to_dict(row)
+
+    @keyword
+    def set_rows(
+        self,
+        data: Union[List, Table],
+        native: bool = False,
+    ) -> List[Union[OrderedDict, Row]]:
```

Comment:
Similar to [this](https://github.com/robocorp/rpaframework/pull/880/files#diff-586ca41af1f200aafdbc7c787d043cc92278cfc019b8a761891814c3fc6088a0R1020).

Reply:
That link opens up all the diffs and I'm not sure what you are referring to.

Reply:
I meant [this](https://github.com/robocorp/rpaframework/pull/880#discussion_r1138210879), just that during the review time you can't get links to past comments till they are posted already.

```python
RowType = Union[OrderedDict, Row]
) -> List[RowType]:
```

Reply:
Ah! good point